File: examples/class-kits/gospider/product_detail.go
package classkit

import scrapyapi "gospider/scrapy"

func NewProductDetailSpider() *scrapyapi.Spider {
	return scrapyapi.NewSpider("product-detail", func(response *scrapyapi.Response) ([]any, error) {
		return []any{
			scrapyapi.NewItem().
				Set("kind", "detail").
				Set("title", response.CSS("title").Get()).
				Set("url", response.URL).
				Set("html_excerpt", response.Text),
		}, nil
	}).AddStartURL("https://example.com/item/123")
}

File: examples/class-kits/rustspider/product_detail.rs
use std::sync::Arc;

use rustspider::scrapy::project as projectruntime;
use rustspider::scrapy::{Item, Output, Response, Spider};

pub fn make_product_detail_spider() -> Spider {
    Spider::new(
        "product-detail",
        Arc::new(|response: &Response| {
            vec![Output::Item(
                Item::new()
                    .set("kind", "detail")
                    .set("title", response.css("title").get().unwrap_or_default())
                    .set("url", response.url.clone())
                    .set("html_excerpt", response.text.clone()),
            )]
        }),
    )
    .add_start_url("https://example.com/item/123")
}

pub fn register() {
    projectruntime::register_spider("product-detail", make_product_detail_spider);
}

